### ISSUE TITLE
Adapt the JWST DQ tests for the Roman pipeline

### ISSUE THREAD

**Issue by stscijgbot-rstdms** [spacetelescope/romancal#119]
_Issue [RCAL-92](https://jira.stsci.edu/browse/RCAL-92) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Adapt the DQ tests from the JWST pipeline to the Roman pipeline.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=518494#comment-518494):_

This work has begun. I wanted to get some tests working in order to test out the DQ Step Code before closing that ticket. The work is available at:

RomanCAL: <https://github.com/PaulHuwe/stcal/tree/RCAL-137_DQviaSTCAL>

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=528080#comment-528080):_

I have implemented the tests as far as I can without support for several datamodels in roman_datamodels. This ticket is blocked until this support is implemented.

NOTE: the RAD work needed is completed in RAD PR-23 spacetelescope/rad#23

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=528422#comment-528422):_

I was able to advance this a bit more, with an assist from [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva), but ran into the block again deep within used STPIPE code. Accordingly, this needs to wait for roman_datamodel implementations mentioned above.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=537234#comment-537234):_

This work is implemented in the following PRs:
#252

spacetelescope/stcal#34

spacetelescope/roman_datamodels#17

spacetelescope/rad#61

There are a few specific parts that are blocked, and are being removed to separate tickets, but the bulk of the work is done here.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=538752#comment-538752):_

Again this cannot be closed until the code developed here is merged and I don't see that. If there is an issue/issues that need to be worked in another ticket to advance this issue that is fine. If we close this issue how do we get the code developed here into the project? Reopening the issue until the PR can be merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=539360#comment-539360):_

Status update on the PRs in this ticket:

RAD 61: This is blocked until INS makes a decision about dynamic dq support.

RDM 17: This is blocked until INS makes a decision about dynamic dq support.

STCAL 34: This is closed, and I will be making a new PR, following the proper STCAL procedure.

RCAL 252: This is blocked until the other three are updated.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=539495#comment-539495):_

Status update on the PRs in this ticket:

RAD 61: This has been unblocked, and work is resuming. (Removing dq_def and addressing PR comments.)

RDM 17: This has been unblocked, and work is resuming. (Removing dq_def and dynamic dq code.)

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=539933#comment-539933):_

The RAD & RDM tickets are back in PR with all requested changes made. Romancal work continues - most of the tests and code are working, I am finalizing the strun step work.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=540389#comment-540389):_

The RAD & RDM PRs have been merged.

Romancal code has a step producing output which is under review.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=540829#comment-540829):_

The romancal work is nor implemented in RCAL PR 262:
#262

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=541720#comment-541720):_

I am addressing a PR comment regarding RampModel instantiation.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=542312#comment-542312):_

All PR comments have been addressed, and ScienceRaw support is now in. This is back in review.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-92?focusedCommentId=543609#comment-543609):_

Approved and merged.
